llama_hub/tools/neo4j_db/base.py

```diff
@@ -0,0 +1,202 @@
+from langchain.schema import (
+    SystemMessage,
+    HumanMessage,
+    AIMessage
+)
+from llama_index import Document
+
+node_properties_query = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE NOT type = "RELATIONSHIP" AND elementType = "node"
+WITH label AS nodeLabels, collect(property) AS properties
+RETURN {labels: nodeLabels, properties: properties} AS output
+
+"""
+
+rel_properties_query = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE NOT type = "RELATIONSHIP" AND elementType = "relationship"
+WITH label AS nodeLabels, collect(property) AS properties
+RETURN {type: nodeLabels, properties: properties} AS output
+"""
+
+rel_query = """
+CALL apoc.meta.data()
+YIELD label, other, elementType, type, property
+WHERE type = "RELATIONSHIP" AND elementType = "node"
+RETURN {source: label, relationship: property, target: other} AS output
+"""
+
+
+def schema_text(node_props, rel_props, rels):
+    return f"""
+  This is the schema representation of the Neo4j database.
+  Node properties are the following:
+  {node_props}
+  Relationship properties are the following:
+  {rel_props}
+  Relationship point from source to target nodes
+  {rels}
+  Make sure to respect relationship types and directions
+  """
+
+
+class Neo4jQueryToolSpec:
+    """
+    This class is responsible for querying a Neo4j graph database based on a provided schema definition.
+    """
+
+    def __init__(self, url, user, password, llm):
```

Comment:
can we use the native llamaindex llm abstractions? they include our native integrations (openai, anthropic, etc.) and also integrate with langchain llms: 

https://gpt-index.readthedocs.io/en/latest/core_modules/model_modules/llms/root.html